test/e2e/specs/editor/blocks/fit-text.spec.js

```diff
@@ -0,0 +1,418 @@
+			await headingBlock.waitFor( { state: 'attached' } );
+
+			const newFontSize = await headingBlock.evaluate( ( el ) => {
+				return window.getComputedStyle( el ).fontSize;
+			} );
+
+			const initialSize = parseFloat( initialFontSize );
+			const newSize = parseFloat( newFontSize );
+
+			// Font size should decrease with more content
+			expect( newSize ).toBeLessThan( initialSize );
+		} );
+
+		test( 'should apply much larger font size with fit text compared to without fit text for a short text', async ( {
+			editor,
+		} ) => {
+			// Insert two paragraphs with same content for comparison
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: {
+					content: 'Hello',
+				},
+			} );
+
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: {
+					content: 'Hello',
+					fitText: true,
+				},
+			} );
+
+			const paragraphBlocks = editor.canvas.locator(
+				'[data-type="core/paragraph"]'
+			);
+
+			// Wait for fit text to apply
+			await paragraphBlocks.nth( 1 ).waitFor( { state: 'attached' } );
+			await expect( paragraphBlocks.nth( 1 ) ).toHaveClass(
+				/has-fit-text/
+			);
+
+			const normalFontSize = await paragraphBlocks
+				.nth( 0 )
+				.evaluate( ( el ) => {
+					return window.getComputedStyle( el ).fontSize;
+				} );
+
+			const fitTextFontSize = await paragraphBlocks
+				.nth( 1 )
+				.evaluate( ( el ) => {
+					return window.getComputedStyle( el ).fontSize;
+				} );
+
+			const normalSize = parseFloat( normalFontSize );
+			const fitTextSize = parseFloat( fitTextFontSize );
+
+			// Fit text should scale up significantly for short content
+			expect( fitTextSize ).toBeGreaterThan( normalSize * 2 );
+		} );
+	} );
+
+	test.describe( 'Frontend functionality', () => {
+		test( 'should render fit text correctly on the frontend', async ( {
+			editor,
+			page,
+		} ) => {
+			await editor.insertBlock( {
+				name: 'core/heading',
+				attributes: {
+					content: 'Frontend Test',
+					level: 2,
+					fitText: true,
+				},
+			} );
+
+			await editor.publishPost();
+
+			const postUrl = await page.evaluate( () =>
+				window.wp.data.select( 'core/editor' ).getPermalink()
+			);
+
+			await page.goto( postUrl );
+
+			const heading = page.locator( 'h2.has-fit-text' );
+
+			await expect( heading ).toBeVisible();
+			await expect( heading ).toHaveClass( /has-fit-text/ );
+
+			// Verify data attribute is set (added by frontend script)
+			const fitTextId = await heading.getAttribute( 'data-fit-text-id' );
+			expect( fitTextId ).toBeTruthy();
+
+			// Verify style element exists for this fit text instance
+			const styleElement = page.locator(
+				`style#fit-text-${ fitTextId }`
+			);
+			await expect( styleElement ).toBeAttached();
+
+			const fontSize = await heading.evaluate( ( el ) => {
+				return window.getComputedStyle( el ).fontSize;
+			} );
+
+			expect( fontSize ).toBeTruthy();
+			expect( parseFloat( fontSize ) ).toBeGreaterThan( 0 );
```

Comment:
Are we testing anything meaningful here? `getComputedStyle` returns a [resolved value](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_cascade/Value_processing#resolved_value), and I can't think of a case in which we won't get a positive in this test.

For example, in this very browser tab's console, querying GitHub, I get:

```js
window.getComputedStyle( document.querySelector('head') ).fontSize // "16px"
window.getComputedStyle( document.querySelector('meta') ).fontSize // "16px"
```

Seems like a better test could be to verify that the computed size property matches the contents of `styleElement`, or at the very least that the size is greater than the document's base font size.

Reply:
You are right this assertion is not meaningful, I updated the test to assert that computed font size matches the contents of styleElement as suggested.